### PR TITLE
feat: add RDMA option to cli args and configs

### PR DESCRIPTION
**Pull request by dsharma-dc**

### CODE REVIEW

**Comment by dsharma-dc** [comment]
bors try

**Comment by dsharma-dc** [comment]
bors try

File: io-engine/src/subsys/config/opts.rs [hunk 1]
```diff
@@ -97,6 +97,8 @@ pub struct NvmfTgtConfig {
     pub opts: NvmfTcpTransportOpts,
```
Comment (tiagolobocastro) [inline io-engine/src/subsys/config/opts.rs:97]: Do we need NvmfRdmaTransportOpts?

**Review by tiagolobocastro — COMMENTED** [review]
We should add rdma feature to the features proto api?

File: io-engine/src/subsys/config/opts.rs (hunk 1 above)
Comment (dsharma-dc) [reply io-engine/src/subsys/config/opts.rs:97]: > Do we need NvmfRdmaTransportOpts?

Yes, that is part of the upcoming changes.

**Comment by dsharma-dc** [comment]
> We should add rdma feature to the features proto api?

Yes, have noted that in the doc. We might need change in Nexus share API too but not likely in first phase.

**Review by dsavitskiy — APPROVED** [review]

**Review by tiagolobocastro — APPROVED** [review]

**Comment by dsharma-dc** [comment]
bors merge

**Pull request merged**